Networking Working Group (#3578)
tock/tock: doc/wg/network/README.md

Network Working Group (NWG)
=================================

- Working Group Charter
- Adopted 7/28/2023

## Goals

The goals of the Network Tock Working Group (NWG) are to:

> - design a set of interfaces (traits) for networking
> - design the user land interfaces for networking
> - define how buffer management should be implemented

Reviewer: ```suggestion
- design a set of interfaces (traits) for networking
- design the user land interfaces for networking
- define how buffer management should be implemented
- determine other kernel infrastructure useful for networking
```

## Members

- Alexandru Radovici, OxidOS Automotive

> - Brandon Ghena, Northwestern University

Reviewer 2: ```suggestion
- Branden Ghena, Northwestern University
```

- Leon Schuermann, Princeton University

> - Tyler Potyondy, UCSD

Reviewer: @tyler-potyondy I wanted to confirm that you're interested in being part of the working group (you can always leave in a couple months if it no longer makes sense to join calls).

- Ioan-Cristian Cirstea, OxidOS Automotive

## Membership and Communication

The networking working group membership is a subset of the people who have
commit (pull request merge) permissions on the Tock repository. It is
intended to be a smaller group that represents the major perspectives
and issues, rather than a complete group. Group membership is decided by
the group: the exact process is not yet determined and may organically
evolve as the group gains momentum.

> The group has a weekly teleconference call. All working group members

Reviewer 2: ```suggestion
The group has a biweekly teleconference call. All working group members
```

Assuming we stick to what we talked about at TockWorld 6. Not strictly opposed to weekly though.

Reviewer: "biweekly" is actually ambiguous in English and means either twice a week or once every two weeks. Both meanings are common which really just means that the word is meaningless.

```suggestion
The group has a teleconference call once every two weeks. All working group members
```

participate in the call. Other people may be invited to participate to
help contribute to particular topics or on-going discussions. The
working group chair decides who beyond the working group members may
participate in the call.

The working group publishes detailed notes of its calls. These will be
posted within a week of a call. This delay is to give participants an
opportunity to correct any errors or better explain points that came up.
They are intended to be a communication mechanism of the group, its

> discussions, the technical issues, and decisions, not a literal
> transcription of what is said.

Reviewer: ```suggestion
discussions, the technical issues, and decisions, not a literal
transcription of what is said.

## Code Purview

At this time, the network working group does not have responsibility for code in any particular directories of Tock. That may change as work progresses, with the possibility of a subdirectory within `capsules/` being created under the purview of this working group.
```

Network Working Group (NWG)
=================================

- Working Group Charter
- Adopted 7/28/2023

## Goals

The goals of the Network Tock Working Group (NWG) are to:

- design a set of interfaces (traits) for networking
- design the user land interfaces for networking
- define how buffer management should be implemented
- determine other kernel infrastructure useful for networking

## Members

- Alexandru Radovici (Chair), OxidOS Automotive
- Branden Ghena, Northwestern University
- Leon Schuermann, Princeton University
- Tyler Potyondy, UCSD
- Ioan-Cristian Cirstea, OxidOS Automotive

## Membership and Communication

The networking working group membership is a subset of the people who have

> commit (pull request merge) permissions on the Tock repository. It is

Reviewer 2: Is this true in practice (I think the answer is now) and do we want this to be the case? I don't see any hard requirement for this.

Reviewer: ```suggestion
The networking working group membership is open to Tock developers interested
in the design of network interfaces. It is
```
What about this?

Reviewer 2: I don‘t see this addressed in the current version.

intended to be a smaller group that represents the major perspectives
and issues, rather than a complete group. Group membership is decided by
the group: the exact process is not yet determined and may organically
evolve as the group gains momentum.

The group has a teleconference call once every two weeks. All working group members
participate in the call. Other people may be invited to participate to
help contribute to particular topics or on-going discussions. The
working group chair decides who beyond the working group members may
participate in the call.

> 

Reviewer 3: ```suggestion
Those looking to engage with the working group are encouraged to join the `#networking-working-group` channel on the [Tock slack](https://github.com/tock/tock/#keep-up-to-date).
```

Reviewer 2: Same here, not yet addressed.

> The working group publishes detailed notes of its calls. These will be

Reviewer 4: ```suggestion

The working group publishes detailed notes of its calls. These will be
```

Reviewer: I don't understand what the suggestion is here. The "suggested change" box has the - and + exactly identical. Unless it's that you added a newline??

Network Working Group (NWG)
=================================

- Working Group Charter

> - Adopted 7/28/2023

Reviewer 4: ```suggestion
- Adopted 8/11/2023
```
